### PR TITLE
Remove experimental jaxpr jvp execution support

### PR DESCRIPTION
**Pull request by astralcai**
**Context:**
This is the first in a staged removal of jaxpr execution support.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-122402]

### CODE REVIEW

File: tests/capture/autograph/test_autograph_item_assignment.py [hunk 1]
```diff
@@ -203,9 +203,10 @@ def circuit(new_val):
     assert jnp.allclose(res, -1.0)
 
     # Test gradient
-    grad = jax.grad(ag_circuit, argnums=0)(new_angle)
-    # d/dx cos(x) = -sin(x), at x=pi, -sin(pi) = 0
-    assert jnp.allclose(grad, 0.0)
+    if not qp.capture.enabled():
+        grad = jax.grad(ag_circuit, argnums=0)(new_angle)
+        # d/dx cos(x) = -sin(x), at x=pi, -sin(pi) = 0
+        assert jnp.allclose(grad, 0.0)
```
Comment (albi3ro) [inline tests/capture/autograph/test_autograph_item_assignment.py]: So this is a capture test, so capture is always going to be enabled. We are also soon going to getting rid of the ability to execute (the `Test forward pass`) part as well. 

I would instead just do:
```
jaxpr = jax.make_jaxpr(ag_circuit)(jnp.pi)
qfunc_jaxpr = jaxpr.eqns[0].params['qfunc_jaxpr']
assert qfunc_jaxpr.eqns[0].invars[0] is qfunc_jaxpr.invars[0]
```
Where we are checking that the input to the RX is the input to the circuit.

File: tests/capture/autograph/test_autograph_item_assignment.py [hunk 2]
```diff
@@ -229,9 +230,10 @@ def circuit(angles, new_val):
     assert jnp.allclose(res, -1.0)
 
     # Test gradient
-    grad = jax.grad(ag_circuit, argnums=1)(angles_in, new_angle)
-    # d/dx cos(x) = -sin(x), at x=pi, -sin(pi) = 0
-    assert jnp.allclose(grad, 0.0)
+    if not qp.capture.enabled():
+        grad = jax.grad(ag_circuit, argnums=1)(angles_in, new_angle)
+        # d/dx cos(x) = -sin(x), at x=pi, -sin(pi) = 0
+        assert jnp.allclose(grad, 0.0)
```
Comment (albi3ro) [inline tests/capture/autograph/test_autograph_item_assignment.py]: For this one, instead of execution I would do something like:
```
jaxpr = jax.make_jaxpr(ag_circuit)(angles_in, jnp.pi)
qfunc_jaxpr = jaxpr.eqns[0].params['qfunc_jaxpr']
assert qfunc_jaxpr.eqns[3].primitive.name == "scatter"
assert qfunc_jaxpr.eqns[4].primitive.name == "slice"
```

This checks we have the right type of primitives for setting a value on an array and then extracting it.

File: tests/capture/gradients/test_capture_diff.py [hunk 3]
```diff
@@ -227,64 +224,6 @@ def func(x):
...
-
-        g_circuit = qp.grad(circuit)
-
-        x = jnp.array([0.5, 0.9])
-        res = g_circuit(x)
-        expected_res = (
-            -jnp.sin(x[0]) * jnp.cos(x[1] ** 2),
-            -2 * x[1] * jnp.sin(x[1] ** 2) * jnp.cos(x[0]),
-        )
-        assert qp.math.allclose(res, expected_res)
```
Comment (albi3ro) [inline tests/capture/gradients/test_capture_diff.py:254]: I would just delete these blocks checking the answers, but then keep all the parts where we are checking the contents of the jaxpr.

File: tests/capture/gradients/test_capture_diff.py [hunk 4]
```diff
@@ -227,64 +224,6 @@ def func(x):
...
-        assert qfunc_jaxpr.eqns[8].primitive == qp.measurements.ExpectationMP._obs_primitive
-
-        assert len(qnode_eqn.outvars) == 1
-        assert qnode_eqn.outvars[0].aval == jax.core.ShapedArray((), fdtype)
-
-        assert len(grad_eqn.outvars) == 1
-        assert grad_eqn.outvars[0].aval == jax.core.ShapedArray((2,), fdtype)
-
-        manual_res = jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, x)
-        assert qp.math.allclose(manual_res, expected_res)
```
Comment (albi3ro) [inline tests/capture/gradients/test_capture_diff.py:286]: This part would also go.

File: tests/capture/gradients/test_capture_diff.py [hunk 5]
```diff
@@ -329,56 +268,6 @@ def func_jax(x):
...
-        assert jaxpr.out_avals == [jax.core.ShapedArray((), fdtype, weak_type=True)] * num_out_avals
-
-        grad_eqn = jaxpr.eqns[0]
-        assert all(invar.aval == in_aval for invar, in_aval in zip(grad_eqn.invars, jaxpr.in_avals))
-        flat_argnums = tuple(
-            [0, 1] * (0 in argnums) + [2] * (1 in argnums) + [3, 4, 5] * (2 in argnums)
-        )
-        diff_eqn_assertions(grad_eqn, scalar_out=True, argnums=flat_argnums, fn=circuit)
-        grad_jaxpr = grad_eqn.params["jaxpr"]
-        assert len(grad_jaxpr.eqns) == 1  # qnode equation
```
Comment (albi3ro) [inline tests/capture/gradients/test_capture_diff.py:373]: Same here. We should keep all of these tests.

File: tests/capture/autograph/test_autograph_item_assignment.py (hunk 1 above)
Comment (astralcai) [reply tests/capture/autograph/test_autograph_item_assignment.py]:
```suggestion
    jaxpr = jax.make_jaxpr(ag_circuit)(jnp.pi)
    qfunc_jaxpr = jaxpr.eqns[0].params['qfunc_jaxpr']
    assert qfunc_jaxpr.eqns[0].invars[0] is qfunc_jaxpr.invars[0]
```

File: tests/capture/autograph/test_autograph_item_assignment.py (hunk 2 above)
Comment (astralcai) [reply tests/capture/autograph/test_autograph_item_assignment.py]:
```suggestion
    jaxpr = jax.make_jaxpr(ag_circuit)(angles_in, jnp.pi)
    qfunc_jaxpr = jaxpr.eqns[0].params['qfunc_jaxpr']
    assert qfunc_jaxpr.eqns[3].primitive.name == "scatter"
    assert qfunc_jaxpr.eqns[4].primitive.name == "slice"
```

File: tests/capture/gradients/test_capture_diff.py (hunk 3 above)
Comment (astralcai) [reply tests/capture/gradients/test_capture_diff.py:254]: Actually I thought the next step was to remove `capture/gradients` completely?

File: tests/capture/autograph/test_autograph_item_assignment.py [hunk 6]
```diff
@@ -202,10 +202,9 @@ def circuit(new_val):
     res = ag_circuit(new_angle)
     assert jnp.allclose(res, -1.0)
 
-    # Test gradient
-    grad = jax.grad(ag_circuit, argnums=0)(new_angle)
-    # d/dx cos(x) = -sin(x), at x=pi, -sin(pi) = 0
-    assert jnp.allclose(grad, 0.0)
+    jaxpr = jax.make_jaxpr(ag_circuit)(jnp.pi)
+    qfunc_jaxpr = jaxpr.eqns[0].params["qfunc_jaxpr"]
+    assert qfunc_jaxpr.eqns[0].invars[0] is qfunc_jaxpr.invars[0]
```
Comment (JerryChen97) [inline tests/capture/autograph/test_autograph_item_assignment.py:207]: Do we still need to check this? I thought that if we already dropped the support for jaxpr jvp exec then we might want to just delete them

File: tests/capture/autograph/test_autograph_item_assignment.py [hunk 7]
```diff
@@ -228,10 +227,10 @@ def circuit(angles, new_val):
     res = ag_circuit(angles_in, new_angle)
     assert jnp.allclose(res, -1.0)
 
-    # Test gradient
-    grad = jax.grad(ag_circuit, argnums=1)(angles_in, new_angle)
-    # d/dx cos(x) = -sin(x), at x=pi, -sin(pi) = 0
-    assert jnp.allclose(grad, 0.0)
+    jaxpr = jax.make_jaxpr(ag_circuit)(angles_in, jnp.pi)
```
Comment (JerryChen97) [inline tests/capture/autograph/test_autograph_item_assignment.py:230]: Same here. Anyways we didn't check the jxpr content in the past; we were executing them but now we don't. So prefer deletion

File: tests/capture/autograph/test_autograph_item_assignment.py (hunk 6 above)
Comment (astralcai) [reply tests/capture/autograph/test_autograph_item_assignment.py:207]: We're dropping support for execution. We still want to capture it.

File: tests/capture/autograph/test_autograph_item_assignment.py (hunk 7 above)
Comment (astralcai) [reply tests/capture/autograph/test_autograph_item_assignment.py:230]: https://github.com/PennyLaneAI/pennylane/pull/9678#discussion_r3421212565

**Review by JerryChen97 — APPROVED** [review]
🎉

File: doc/releases/changelog-dev.md [hunk 8]
```diff
@@ -491,6 +491,9 @@
 * Plxpr transforms have been removed.
   [(#9637)](https://github.com/PennyLaneAI/pennylane/pull/9637)
 
+* PLxPR execution support have been removed for `default.qubit`.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:494]:
```suggestion
* PLxPR gradient execution support has been removed for `default.qubit`.
```

**Review by albi3ro — APPROVED** [review]
🚀
